extension/src/XliffEditor/XliffEditorPanel.ts

```diff
@@ -98,28 +101,31 @@ export class XliffEditorPanel {
                         }
                         return;
                     case "update":
+                        let suggestedTranslations: { id?: string, targetText: string }[] = [];
                         vscode.window.showInformationMessage(message.text);
+                        let targetUnit = this._xlfDocument.getTransUnitById(message.transunitId);
                         this._xlfDocument.getTransUnitById(message.transunitId).targets[0].textContent = message.targetText;
                         this._xlfDocument.getTransUnitById(message.transunitId).targets[0].translationToken = undefined;
+                        this._xlfDocument.getTransUnitById(message.transunitId).insertCustomNote(CustomNoteType.RefreshXlfHint, "Translated with Xliff Editor");
+                        this._xlfDocument.transunit.filter(a => (a.source === targetUnit.source) && !a.identicalTargetExists(message.targetText) && (a.id !== targetUnit.id) /*&& a.hasCustomNote(CustomNoteType.RefreshXlfHint)*/).forEach(unit => {
+                            let suggestion = new Target(message.targetText);
+                            suggestion.translationToken = TranslationToken.Suggestion;
+                            if (unit.targets.length === 1 && (unit.targets[0].textContent === "")) {
+                                unit.targets[0] = suggestion;
+                                suggestedTranslations.push({ id: unit.id, targetText: suggestion.textContent });
+                            }
+                        });
                         this._xlfDocument.toFileSync(this._xlfDocument._path);
-
-                        return;
-                    case "filter":
-                        if (message.text === "review") {
-                            let filteredXlf = new Xliff(
-                                this._xlfDocument.datatype,
-                                this._xlfDocument.sourceLanguage,
-                                this._xlfDocument.targetLanguage,
-                                this._xlfDocument.original
-                            );
-                            filteredXlf._path = this._xlfDocument._path;
-                            filteredXlf.transunit = this._xlfDocument.transunit.filter(u => u.targets[0].translationToken !== undefined);
-                            this._update(filteredXlf);
-
-                        } else if (message.text === "all") {
+                        if (suggestedTranslations.length > 0) {
+                            this._panel.webview.postMessage({ command: "suggestions", suggestions: suggestedTranslations });
```

Comment:
I cannot see the command suggestions being handled anywhere...? but maybe that was intentionally left out for now?

Reply:
Correct, this does not belong here anymore. Clean up fail :)